modules/core/src/main/scala/org/tessellation/infrastructure/trust/TrustStorageUpdater.scala

```diff
@@ -0,0 +1,39 @@
+package org.tessellation.infrastructure.trust
+
+import cats.effect.std.Supervisor
+import cats.effect.{Async, Ref}
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import scala.concurrent.duration.DurationInt
+
+import org.tessellation.domain.trust.storage.TrustStorage
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.gossip.Gossip
+
+trait TrustStorageUpdater[F[_]] {
+  def update: F[Unit]
+}
+
+object TrustStorageUpdater {
+
+  def make[F[_]: Async](
+    getOrdinal: Ref[F, Option[SnapshotOrdinal]],
+    gossip: Gossip[F],
+    storage: TrustStorage[F]
+  ): TrustStorageUpdater[F] = new TrustStorageUpdater[F] {
+    override def update: F[Unit] = for {
+      maybeOrdinal <- getOrdinal.get
+      _ <- maybeOrdinal.traverse_(storage.updateCurrent) // This must come first.
+      maybeOrdinalPublicTrust <- maybeOrdinal.flatTraverse(storage.updateNextFromInternal)
+      _ <- maybeOrdinalPublicTrust.traverse_(gossip.spread(_))
+    } yield ()
+  }
+
+  def daemon[F[_]: Async: Supervisor](ac: TrustStorageUpdater[F]): Daemon[F] =
```

Comment:
what is `ac`?

Reply:
I'll defer to @sdavidp on this.

Reply:
🤷   looks like a variable name left over from my example code

Reply:
@marcinwadon , what would you like this variable name to be?

Reply:
to reflect what the variable means (or a short of it) :)

Reply:
Changed `ac` to `updater`.